core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -280,30 +280,33 @@ object AdminUtils extends Logging with AdminUtilities {
           s"Assignment: $existingAssignment"))
 
     val partitionsToAdd = numPartitions - existingAssignment.size
-    if (partitionsToAdd <= 0)
+    if (partitionsToAdd < 0)
       throw new InvalidPartitionsException(
         s"The number of partitions for a topic can only be increased. " +
           s"Topic $topic currently has ${existingAssignment.size} partitions, " +
           s"$numPartitions would not be an increase.")
+    else if (partitionsToAdd == 0) {
+      existingAssignment
```

Comment:
I think I wasn't totally clear here. I meant that we should throw an exception here, same as we don't allow `createTopics` to be called if the topic already exists. This is the behaviour you originally implemented, but then I asked you to change it. So sorry about that.

Reply:
Ah, I misunderstood indeed. Should now be fixed.